### PR TITLE
fix race condition caused by writing to entry.Data

### PR DESCRIPTION
**Pull request by h12w**
This PR should be able to fix https://github.com/sirupsen/logrus/issues/468, and it uses the same technique as JSONFormatter.

### CODE REVIEW

File: text_formatter.go [hunk 1]
```diff
@@ -160,7 +163,7 @@ func (f *TextFormatter) Format(entry *Entry) ([]byte, error) {
 		timestampFormat = defaultTimestampFormat
 	}
 	if f.isColored() {
-		f.printColored(b, entry, keys, timestampFormat)
+		f.printColored(b, entry, keys, nil, timestampFormat)
```
Comment (dgsb) [inline text_formatter.go]: Should we give the new data array as parameter here ?

Comment (dgsb) [inline text_formatter.go]: Should we give the new data array as parameter here ?

Comment (h12w) [reply text_formatter.go]: Fixed.

Comment (h12w) [reply text_formatter.go]: Fixed.

**Pull request merged**